### PR TITLE
georeg / rm80662 / add selfie step into signup wizard

### PR DESCRIPTION
**Pull request by george-usynin-binary**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx [hunk 1]
```diff
@@ -28,9 +50,13 @@ const SelfieUpload = ({ initial_values, goBack, onConfirm, onFileDrop }) => {
                     return (
                         <Form className={`${ROOT_CLASS}__form`}>
                             <div className={`${ROOT_CLASS}__fields-content`}>
-                                <Text as='h3' size='s' weight='bold' color='prominent'>
-                                    {localize('Upload your selfie')}
-                                </Text>
+                                {is_pa_signup ? (
+                                    selfie_header_for_pa_signup
```
Comment (farzin-deriv) [inline packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx:54]: @george-usynin-binary Let's render it as a component and also wrap it with `useMemo`.

File: packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx [hunk 2]
```diff
@@ -20,6 +39,9 @@ const SelfieUpload = ({ initial_values, goBack, onConfirm, onFileDrop }) => {
                 initialValues={initial_values || setInitialValues([SELFIE_DOCUMENT])}
                 validate={values => validateFields(values, undefined, [SELFIE_DOCUMENT])}
                 onSubmit={onConfirm}
+                innerRef={formik_actions =>
+                    formik_actions ? setFormikValues(formik_actions.values) : setFormikValues({})
+                }
```
Comment (farzin-deriv) [inline packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx:44]:
```suggestion
                innerRef={formik_actions => setFormikValues(formik_actions?.values || {})}
```

Comment (farzin-deriv) [reply packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx:44]: Also, you need to do this setState conditionally because if it's not `is_pa_signup` then you are creating unnecessary rendering.

File: packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx [hunk 3]
```diff
@@ -9,7 +9,26 @@ import Uploader from './uploader.jsx';
 import { setInitialValues, validateFields } from './utils';
 import { ROOT_CLASS, SELFIE_DOCUMENT } from '../constants';
 
-const SelfieUpload = ({ initial_values, goBack, onConfirm, onFileDrop }) => {
+const SelfieUpload = ({ initial_values, is_pa_signup, goBack, onConfirm, onFileDrop, setIsSelfieStepEnabled }) => {
+    const [formik_values, setFormikValues] = React.useState({});
+
+    React.useEffect(() => {
+        if (typeof setIsSelfieStepEnabled === 'function') setIsSelfieStepEnabled(formik_values[SELFIE_DOCUMENT.name]);
```
Comment (farzin-deriv) [inline packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx:16]:
```suggestion
        setIsSelfieStepEnabled?.(formik_values[SELFIE_DOCUMENT.name]);
```
@george-usynin-binary Do we need to check the type? 🤔

File: packages/cashier/globals.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module '@deriv/account';
```
Comment (farzin-deriv) [inline packages/cashier/globals.d.ts:1]: @george-usynin-binary Please remove this file 🙏🏻

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx [hunk 5]
```diff
@@ -16,6 +17,9 @@ const SignupWizard = ({ closeWizard }: TSignupWizardProps) => {
     const [is_cancel_wizard_dialog_active, setIsCancelWizardDialogActive] = React.useState(false);
     const [current_step_key, setCurrentStepKey] = React.useState<string>();
     const [is_country_selected, setIsCountrySelected] = React.useState(false);
+    const [selfie, setSelfie] = React.useState<{ selfie_with_id: TSelfie }>();
+    const [is_selfie_step_enabled, setIsSelfieStepEnabled] = React.useState(false);
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:21]: @george-usynin-binary It's not required in this PR, Maybe you can just leave a todo for now, Since the number of states will grow in this parent component we can use `useReducer` hook instead of `useState`.

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard-steps/selfie-step/selfie-step.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    setIsSelfieStepEnabled: (value: boolean) => void;
+    setSelfie: (value: { selfie_with_id: TSelfie }) => void;
+};
+
+const SelfieStep = ({ selfie, setIsSelfieStepEnabled, setSelfie }: TSelfieStep) => {
+    const onFileDrop = (value: TSelfie) => setSelfie({ selfie_with_id: value });
+
+    //TODO: change the description for the selfie depending on the step number
+    return (
+        <>
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard-steps/selfie-step/selfie-step.tsx:41]: @george-usynin-binary Is the fragment needed?

File: packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx (hunk 1 above)
Comment (george-usynin-binary) [reply packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx:54]: @farzin-deriv sure, will move it to a separate component

File: packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx (hunk 3 above)
Comment (george-usynin-binary) [reply packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx:16]: @farzin-deriv  believe that we can use optional chaining

File: packages/cashier/globals.d.ts (hunk 4 above)
Comment (george-usynin-binary) [reply packages/cashier/globals.d.ts:1]: @farzin-deriv if i remove this file i got an error when i try to import components from the account package
Screenshot 2022-11-24 at 15 13 57

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 5 above)
Comment (george-usynin-binary) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:21]: @farzin-deriv yeah we can refactor it to userReducer hook, but i don't realise why you want to remove this setState for now. Why it's not required now?

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard-steps/selfie-step/selfie-step.tsx (hunk 6 above)
Comment (george-usynin-binary) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard-steps/selfie-step/selfie-step.tsx:41]: @farzin-deriv yes it's redundant, will remove it

File: packages/cashier/globals.d.ts (hunk 4 above)
Comment (george-usynin-binary) [reply packages/cashier/globals.d.ts:1]: we have this workaround in other packages, maybe there is a better solution to remove the error?

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 5 above)
Comment (george-usynin-binary) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:21]: @farzin-deriv i refactored code with `useReducer` hook, could you pls have a look :pray::skin-tone-2:

File: packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx (hunk 1 above)
Comment (farzin-deriv) [reply packages/account/src/Components/poi/status/unsupported/card-details/selfie-upload.jsx]: @george-usynin-binary Yes I meant `React.memo` 😅
Although I see you have moved it outside of the `SelfieUpload` component, So it won’t be created in each rerender and `React.memo` is not necessary anymore.

File: packages/cashier/globals.d.ts (hunk 4 above)
Comment (farzin-deriv) [reply packages/cashier/globals.d.ts:1]: @george-usynin-binary It's okay to have the error, We will gradually fix the errors in packages as we are migrating to TS.

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 5 above)
Comment (farzin-deriv) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx]: @george-usynin-binary By `not required` I meant the change I'm asking for is not required to be done in your PR and a todo would be enough 😅 Sorry for the misunderstanding 🙏🏻

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/steps-reducer.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    return {
+        type: ACTION_TYPES.SET_SELFIE_STEP_ENABLED,
+        value,
+    };
+};
+
+// Initial state
+export const initial_state = { selfie: null, is_selfie_step_enabled: false };
+
+// Reducer
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/steps-reducer.ts:33]: @george-usynin-binary We can simplify and isolate this reducer, Let’s have a call and do it in another PR 🙇🏻

**Review by farzin-deriv — COMMENTED** [review]
@george-usynin-binary Just remove the declaration file, The rest looks good 🚀

File: packages/cashier/tsconfig.json [hunk 8]
```diff
@@ -16,4 +16,4 @@
         "baseUrl": "./"
     },
     "include": ["src"]
-}
+}
```
Comment (farzin-deriv) [inline packages/cashier/tsconfig.json:19]: @george-usynin-binary Missing EOF 😬

**Review by farzin-deriv — APPROVED** [review]

**Review by hamid-deriv — APPROVED** [review]
